src/api/java/com/minecolonies/api/compatibility/CompatibilityManager.java

```diff
@@ -447,12 +455,23 @@ public ItemStack getRandomLuckyOre()
     {
         if (random.nextInt(ONE_HUNDRED_PERCENT) <= Configurations.gameplay.luckyBlockChance)
         {
-            Collections.shuffle(luckyOres);
-            return luckyOres.get(0).getItemStack().copy();
+            return luckyOres.get(random.nextInt(luckyOres.size())).getItemStack().copy();
         }
         return ItemStack.EMPTY;
     }
 
+    @Override
+    public Tuple<String, Integer> getRandomEnchantment(final int buildingLevel)
+    {
+        final List<Tuple<String, Integer>> list = enchantments.getOrDefault(buildingLevel, new ArrayList<>());
+        if (list.isEmpty())
+        {
+            return new Tuple<>("protection", 1);
```

Comment:
This should return something else if no enchantment can be found for a given Level.
I would prefer it if this method returned an Optional instead.

You could also consider turning the Tuple<String, Integer> into a Util class that takes the enchantment and the Level and allows you to directly put it on an itemstack. Instead of returning the data here only.